tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc

```diff
@@ -8074,8 +8074,8 @@ TEST_F(EnergyPlusFixture, VRFTU_CalcVRFSupplementalHeatingCoilWater)
     state->dataWaterCoils->WaterCoil.allocate(state->dataWaterCoils->NumWaterCoils);
     state->dataWaterCoils->WaterCoil(CoilNum).Name = thisVRFTU.SuppHeatCoilName;
     state->dataWaterCoils->WaterCoil(CoilNum).WaterCoilType_Num = state->dataWaterCoils->WaterCoil_SimpleHeating;
-    state->dataWaterCoils->WaterCoil(CoilNum).WaterCoilModel = state->dataWaterCoils->CoilType_Heating;
-    state->dataWaterCoils->WaterCoil(CoilNum).WaterCoilType = state->dataWaterCoils->CoilType_Heating;
+    state->dataWaterCoils->WaterCoil(CoilNum).WaterCoilModel = state->dataWaterCoils->CoilModel_Cooling;
+    state->dataWaterCoils->WaterCoil(CoilNum).WaterCoilType = HVACControllers::iCoilType::Heating;
```

Comment:
@mjwitte I think it's worth drawing your attention to this particular change. I changed ```WaterCoilType``` here to use an enum value instead of const ints. In doing so, I found that in this unit test, ```WaterCoilModel``` was using the "CoilType" int const value instead of the "CoilModel" int const values at this location. In order to replicate the same behavior, ```CoilType_Heating``` which had a value of 2 had to be moved to ```CoilModel_Cooling```, which also has a value of 2. So now this unit test looks a funny, with CoilType values being set to "Heating" and the CoilModel value being set to "Cooling".

I suppose the main question I have is whether you think this looks like just a quirk of this unit test, or if this points to a larger issue that needs to be invetigated?

Reply:
@mitchute No, that's bad.  Why is there a coil type enum class in `HVACControllers` now? 

Earlier in this same unit test a different coil type is set which is probably the source of the inconsistency.
https://github.com/NREL/EnergyPlus/blob/b7b900ecde5abcfdaa232726a85f5b2dd12590f0/tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc#L8058

And why does `dataWaterCoils->WaterCoil` have two different type variables, `WaterCoilType_Num` and `WaterCoilType`?

But you had to change `WaterCoilModel` to cooling? Now I'm really confused.

Reply:
Here's a high level comment. EnergyPlus and its API should deal with strings as little as possible. Enumerated values should not be strings, they should be enums. That's what the keyword `enum` means. The only things that should be strings are (dynamic) names. This is probably not the right time to do this sweep, but a sweep is needed.

Reply:
Let's see if I can recreate the chain of events that lead to this:
1. Change HVACController::CoilType* const ints to enum class.
2. Those are checked against ```iNodeType``` which is set by ```WaterCoils::CheckActuatorNode```.
3. Internally, ```CheckActuatorNode``` sets ```iNodeType``` from ```WaterCoils::WaterCoil(coilNum).WaterCoilType```.
4. Changing ```WaterCoilType``` from ```int``` to ```HVACControllers::iCoilType``` then resulted in noticing this problem.

No idea why we have two different type variables.

Reply:
<sigh> I see, `WaterCoilType` is only heating or cooling, `WaterCoilType_Num` is
```
        int const WaterCoil_SimpleHeating;
        int const WaterCoil_DetFlatFinCooling;
        int const WaterCoil_Cooling;
```
So, `WaterCoilType` could be combined with the other with some OR logic added where needed. FFR I guess.
But `WaterCoilModel` is essentially the same list of options?
```
        int const CoilModel_Simple; <-- which is used for WaterCoil_SimpleHeating
        int const CoilModel_Cooling;
        int const CoilModel_Detailed;
```
So, it looks like `WaterCoil(CoilNum).WaterCoilModel` and `WaterCoil(CoilNum).WaterCoilType_Num` are redundant and one or the other should be removed. At a minimum, they should bot work off the same enum class.

Reply:
@Myoldmopar @mitchute We still have these lines in this unit test.
```
 state->dataWaterCoils->WaterCoil(CoilNum).WaterCoilModel = state->dataWaterCoils->CoilModel_Cooling;
    state->dataWaterCoils->WaterCoil(CoilNum).WaterCoilType = HVACControllers::iCoilType::Heating;
```
What's the plan for fixing this?

Reply:
What is the semantic distinction between these two variables? Why are they both necessary?

Reply:
They aren't really necessary. One has choices of "heating" or "cooling" and the other has choices of "heating", "cooling-simple" and "cooling-detailed". With a few logic changes the binary one could be removed. See https://github.com/NREL/EnergyPlus/pull/8615#discussion_r593356185

Reply:
Alternatively replace all names with `auto` and let the compiler figure it out.